src/sentry/tasks/backfill_seer_grouping_records.py

```diff
@@ -159,64 +159,63 @@ def backfill_seer_grouping_records(
     events_entity = Entity("events", alias="events")
     redis_client = redis.redis_clusters.get(settings.SENTRY_MONITORS_REDIS_CLUSTER)
 
-    query = Query(
-        match=events_entity,
-        select=[
-            Column("group_id"),
-            Function("max", [Column("event_id")], "event_id"),
-        ],
-        groupby=[Column("group_id")],
-        where=[
-            Condition(Column("project_id"), Op.EQ, project.id),
-            Condition(Column("group_id"), Op.IN, group_id_batch),
-            Condition(
-                Column("timestamp", entity=events_entity), Op.GTE, time_now - timedelta(days=90)
-            ),
-            Condition(Column("timestamp", entity=events_entity), Op.LT, time_now),
-        ],
-        orderby=[OrderBy(Column("group_id"), Direction.ASC)],
-    )
-
-    request = Request(
-        dataset=Dataset.Events.value,
-        app_id=Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value,
-        query=query,
-        tenant_ids={
-            "referrer": Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value,
-            "cross_org_query": 1,
-        },
-    )
+    # TODO(jangjodi): Only query per group if it has over 1 million events
+    snuba_event_data, group_id_batch_all = [], copy.deepcopy(group_id_batch)
+    for group_id in group_id_batch_all:
+        query = Query(
+            match=events_entity,
+            select=[
+                Column("group_id"),
+                Column("event_id"),
+            ],
+            where=[
+                Condition(Column("project_id"), Op.EQ, project.id),
+                Condition(Column("group_id"), Op.EQ, group_id),
+                Condition(
+                    Column("timestamp", entity=events_entity), Op.GTE, time_now - timedelta(days=90)
+                ),
+                Condition(Column("timestamp", entity=events_entity), Op.LT, time_now),
```

Comment:
You can make this query much more efficient by doing `group.last_seen - 5 mins <= timestamp < group.last_seen + 5 mins`

Potentially, you could also make the query a lot faster for batched groups by doing something like
```
(group_id = 123 and timestamp < <group_123_last_seen> + 5 mins and timestamp >= <group_123_last_seen> - 5 mins)
or (group_id = 456 and timestamp < <group_456_last_seen> + 5 mins and timestamp >= <group_456_last_seen> - 5 mins)
or ...

```

Reply:
Thank you! I've modified the query to limit it to the 10 minutes around `group.last_seen` and I'll follow up with another PR for trying to do it for batched groups